libs/vkd3d/device.c

```diff
@@ -4311,18 +4312,131 @@ static HRESULT STDMETHODCALLTYPE d3d12_device_CreateSharedHandle(d3d12_device_if
+    }
+
+    FIXME("Creating shared handle for type of object %p unsupported\n", object);
     return E_NOTIMPL;
 }
 
 static HRESULT STDMETHODCALLTYPE d3d12_device_OpenSharedHandle(d3d12_device_iface *iface,
         HANDLE handle, REFIID riid, void **object)
 {
-    FIXME("iface %p, handle %p, riid %s, object %p stub!\n",
+    struct d3d12_device *device = impl_from_ID3D12Device(iface);
+    HRESULT hr;
+
+    TRACE("iface %p, handle %p, riid %s, object %p\n",
             iface, handle, debugstr_guid(riid), object);
 
+    if (IsEqualGUID(riid, &IID_ID3D12Resource))
+    {
+        struct DxvkSharedTextureMetadata metadata;
+        D3D12_RESOURCE_DESC1 desc;
+        D3D12_HEAP_PROPERTIES heap_props;
+        struct d3d12_resource *resource;
+        bool kmt_handle = false;
+
+        if ((ULONG_PTR)handle & 0x40000000 && ((ULONG_PTR)handle - 2) % 4 == 0)
+        {
+            handle = vkd3d_open_kmt_handle(handle);
+            kmt_handle = true;
+
+            if (handle == INVALID_HANDLE_VALUE)
+            {
+                WARN("Failed to open KMT-style ID3D12Resource shared handle.\n");
+                *object = NULL;
+                return E_INVALIDARG;
+            }
+        }
+
+        if (!vkd3d_get_shared_metadata(handle, &metadata, sizeof(metadata), NULL))
+        {
+            WARN("Failed to get ID3D12Resource shared handle metadata.\n");
+            if (kmt_handle)
+                CloseHandle(handle);
+
+            *object = NULL;
+            return E_INVALIDARG;
+        }
+
+        desc.Dimension = D3D12_RESOURCE_DIMENSION_TEXTURE2D;
+        desc.Alignment = 0;
+        desc.Width = metadata.Width;
+        desc.Height = metadata.Height;
+        desc.DepthOrArraySize = metadata.ArraySize;
+        desc.MipLevels = metadata.MipLevels;
+        desc.Format = metadata.Format;
+        desc.SampleDesc = metadata.SampleDesc;
+        switch (metadata.TextureLayout)
+        {
+            case D3D11_TEXTURE_LAYOUT_UNDEFINED: desc.Layout = D3D12_TEXTURE_LAYOUT_UNKNOWN; break;
+            case D3D11_TEXTURE_LAYOUT_ROW_MAJOR: desc.Layout = D3D12_TEXTURE_LAYOUT_ROW_MAJOR; break;
+            case D3D11_TEXTURE_LAYOUT_64K_STANDARD_SWIZZLE: desc.Layout = D3D12_TEXTURE_LAYOUT_64KB_STANDARD_SWIZZLE; break;
+            default: desc.Layout = D3D12_TEXTURE_LAYOUT_UNKNOWN;
+        }
+        desc.Flags = D3D12_RESOURCE_FLAG_ALLOW_SIMULTANEOUS_ACCESS;
+        if (metadata.BindFlags & D3D11_BIND_RENDER_TARGET)
+            desc.Flags |= D3D12_RESOURCE_FLAG_ALLOW_RENDER_TARGET;
+        desc.SamplerFeedbackMipRegion.Width = 0;
+        desc.SamplerFeedbackMipRegion.Height = 0;
+        desc.SamplerFeedbackMipRegion.Depth = 0;
+
+        heap_props.Type = D3D12_HEAP_TYPE_DEFAULT;
+        heap_props.CPUPageProperty = D3D12_CPU_PAGE_PROPERTY_UNKNOWN;
+        heap_props.MemoryPoolPreference = D3D12_MEMORY_POOL_UNKNOWN;
+        heap_props.CreationNodeMask = 0;
+        heap_props.VisibleNodeMask = 0; // ?
```

Comment:
What is your question? Looks fine to me.

also no // :frog:

Reply:
Ah yeah forgot to remove that.  At first I read it wrong and thought it had something to do with what command queues the resource could be used in.